util/patch/patch_test.go

```diff
@@ -1896,6 +1917,18 @@ func TestPatchHelperForV1beta2Transition(t *testing.T) {
 			t.Log("Patching the object")
 			g.Expect(patcher.Patch(ctx, obj)).To(Succeed())
 
+			// Adding Available as a second condition, but it should be sorted as first
+			t.Log("Creating a new patch helper")
+			patcher, err = NewHelper(obj, env)
+			g.Expect(err).ToNot(HaveOccurred())
+
+			t.Log("Marking condition and back compatibility condition Available=True")
+			conditions.MarkTrue(obj, clusterv1.ReadyCondition)
+			v1beta2conditions.Set(obj, metav1.Condition{Type: "Available", Status: metav1.ConditionTrue, Reason: "AllGood", LastTransitionTime: now})
```

Comment:
This is now mixed, the old Ready condition and the new Available condition. Intended?

Reply:
Good catch, it was a copy/paste left over.
Dropped now (we are testing sort only for v1beta2 conditions)